core/shared/src/main/scala/matryoshka/Delay.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2014–2016 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package matryoshka
+
+import scalaz._
+
+/** To avoid diverging implicits with fixed-point types, we need to defer the
+  * lookup.
+  */
+trait Delay[F[_], G[_]] {
+  def apply[A](fa: F[A]): F[G[A]]
+}
+
+object Delay {
+  implicit def fromNT[F[_], G[_]](nt: F ~> (F ∘ G)#λ): Delay[F, G] =
```

Comment:
I assume this is implicit to avoid having to rewrite all the Delay instances? Maybe a comment that we’d like this to be _explicit_ in the future?